nilearn/plotting/matrix_plotting.py

```diff
@@ -62,11 +62,40 @@ def plot_matrix(mat, title=None, labels=None, figure=None, axes=None,
         grid : color or False, optional
             If not False, a grid is plotted to separate rows and columns
             using the given color.
+        reorder : boolean or {'single', 'complete', 'average'}, optional
+            If not False, reorders the matrix into blocks of clusters.
+            Accepted linkage options for the clustering are 'single',
+            'complete', and 'average'. True defaults to average linkage.
         kwargs : extra keyword arguments
             Extra keyword arguments are sent to pylab.imshow
 
         Returns Matplotlib AxesImage instance
     """
+    if reorder is not False:
+        if labels is None or labels is False:
+            raise ValueError("Labels are needed to show the reordering.")
+        try:
+            from scipy.cluster.hierarchy import (linkage, optimal_leaf_ordering,
+                                                 leaves_list)
+        except ImportError:
```

Comment:
This has to be enforced in the dependences of the apckage, not a the level of individual functions.

Reply:
I disagree: the error is raised only of the user tries to reorder, which is not on by default.